### ISSUE TITLE
pandas dataframe.head(n) should have proper output

### ISSUE THREAD

**Issue by antonkulaga** [polynote/polynote#836]
I already mentioned big usability problems for spark DataFrame visualizations ( https://github.com/polynote/polynote/issues/715 ). However, for non-spark pandas use-cases they are even more relevant when I have non-spark pandas data frame, I expect that:
```
df.head(10)
```
will show me first 10 rows of the dataframe in the cell's output section instead of showing me a link which I should click to get a popup-like window which will block scrolling of my notebook and also requires +1 click to see the content.
When I look at jupyter notebooks of my collegues I use outputs of the cells df.head(number) as clues for understanding of how the data is processed. Polynote makes it a nightmare to read such cells as just to see several rows in each of the cells I have to click two times!

**Comment by jeremyrsmith** [comment]
So, I think the issue here is that `df.head(10)` gives you an `Array[Row]` which presents as a `StreamingDataRepr` because it's a `Seq`. What we need to do here is differentiate between `StreamingDataRepr` that is lazily evaluated (like `DataFrame`) and that can be eagerly pushed (like `Array[Row]`) so that we can minimize UI clicks to see things that you know you should be able to see.

@antonkulaga Just to provide some context, Polynote has been optimized with "WAY TOO BIG" data in mind, because that's what we're used to. So the backend will basically *never* push data to the front-end, unless it's specifically been aggregated via the plotting UI. This is because unaggregated data (in our use cases) is almost guaranteed to overwhelm your browser. Clearly, this isn't a common use case so we need to be able identify when it's OK to eagerly send stuff to the client.

**Comment by jeremyrsmith** [comment]
Oh, actually I missed the pandas part of this. That makes it even more complicated 😬 but definitely something we should try to support. Thanks for the report @antonkulaga

**Comment by antonkulaga** [comment]
well, it makes it easier, you can check if spark is configured by the notebook and if it is not - then stop treating pandas dataframes as if they are spark ones.
While for spark my suggestion is that you have to go from two-clicks mode to one-click mode, just display a button "show output" in the outputsection of the dataframe and if the user clicks show it in the output section instead of displaying ugly popup-like window that blocks scrolling of the notebook. Or if some people like that window - just display a separate button: show dataframe as popup (actually a proper popup - as a new browser window would be more usable than a window in the middle of notebook that blocks using it while you are looking at the dataframe data).

**Comment by raproth** [comment]
@antonkulaga Yes I would also support that, think the separate window is a pain to use. Polynote should be able to render simple tables. I liked the way table widgets works for jupyter and spark-notebook (rendered as a cell output, maybe even with sorting and pagination). For DataFrames, the table should only be rendered with an additional button click (because they are lazy), Seq`s should be rendered without clicks
